dev/RadioButtons/RadioButtons.cpp

```diff
@@ -61,6 +62,18 @@ void RadioButtons::OnListViewSelectionChanged(const winrt::IInspectable& sender,
     m_selectionChangedEventSource(sender, args);
 }
 
+void RadioButtons::OnListViewGettingFocus(const winrt::IInspectable& /*sender*/, const winrt::GettingFocusEventArgs& args)
```

Comment:
>OnListViewGettingFocus [](start = 19, length = 22)

This will get called when getting focus via any mechanism - pointer, keyboard, gamepad, etc.

If you only want to do this for tabbing, this should check for args.InputDevice == FocusInputDeviceKind.Keyboard

I think when getting focus from a gamepad, for example, this would be a bad experience since it would cause a selection just to explore the UI.  I guess I'd think the same for keyboard, so the whole change is confusing to me, but at minimum for gamepad it seems incorrect.

Reply:
I'm still unclear on why we want this at all - I tried making a WPF app with a list of radio buttons and when I tab through it, no radio button gets auto-selected.  What's the rationale for wanting this?  It seems to confuse selection and focus exploration - which may not be great for keyboard-first users and I would imagine could impact accessibility.

---
In reply to: [279900125](https://github.com/microsoft/microsoft-ui-xaml/pull/635#discussion_r279900125) [](ancestors = 279900125)

Reply:
@kikisaints this one is yours. Can you or @jevansaks  comment?

Reply:
@Austin-Lamb The original logic was that since tabbing would move you to the first selected item, if there is one selected, then it would be weird if there were no items previously selected, to have tabbing into it not select anything upon initial focus.

However, in creating a WinForms app, which is what accessibility was basing most of their asks on for this feature - you're right, tabbing into a group of radiobuttons (with nothing yet selected/checked) does not put focus on the initial tabbed one, only once arrow-ing has started.

It's also been a while since I've reviewed this with the accessibility team and Chigusa. This may have been an ask directly from accessibility to have this. @chigy do you remember?

If I'm mistaken, then I update the spec, but I'll also sync with accessibility. As I want to make sure this isn't an "upgrade" feature that they asked for that differed from the older WPF/Win32 functionality and we might be skipping out on if we don't do it.

Reply:
@kikisaints , while I don't remember the past conversation, somewhere in there I feel like there was a discussion of putting focus on the whole entire RBG, but not sure if that's a radical change. Here's what I feel. Having no indication of focus appearing is bad because it goes against the rule of always making focus visible, however forcing selection because of that desire is bad. If I have to pick one of the two bad options, I'd go against forcing selection because user will not be able to undo it. Not sure if this answers the question...?

Reply:
Why wouldn't we just have a focus rect draw around each radio button as you tab around, like it does in WPF?  Then you'd know where focus is, can hit spacebar to select if you wish, or keep tabbing if you want to continue exploring?  That's how it works in WPF, and makes sense to me as a keyboard user.

Reply:
Sounds like we'd prefer the WPF/Win32 "tab into radiobutton group" behavior, which is if there is nothing selected, then it should not select upon initial tabbing into the group. However, regular selection-follows-focus will continue once using the arrow keys.

Reply:
> Why wouldn't we just have a focus rect draw around each radio button as you tab around, like it does in WPF? Then you'd know where focus is, can hit spacebar to select if you wish, or keep tabbing if you want to continue exploring? That's how it works in WPF, and makes sense to me as a keyboard user.

Unfortunately this does not work for our accessibility users, and that behavior did not meet their approval when reviewing how to improve the keyboarding behavior with radiobuttons in a group.

To them, the WinForms/Win32 behavior was best. Which only allowed tabbing "into" and "out of" the group. However navigation within was done purely with the arrow keys. We explored the WPF behavior with them and the consensus was that it did not feel good.